### ISSUE TITLE
Results parameter passed to formatter isn't correct

### ISSUE THREAD

**Issue by hsxfjames** [eslint/eslint#8890]
**Tell us about your environment**

* **ESLint Version: 4.1.1**
* **Node Version: v8.1.3**
* **npm Version: 5.1.0**

**What parser are you using?**
babel-eslint

**Please show your full configuration:**
<details>
<summary>Configuration</summary>

```js
module.exports = {
    root: true,
    parser: 'babel-eslint',
    parserOptions: {
        ecmaVersion: 6,
        sourceType: 'module',
        ecmaFeatures: {
            'impliedStrict': true
        }
    },
    env: {
        browser: true,
    },
    plugins: [
        'html'
    ],
    'rules': {
        'space-before-blocks': 2,
        'indent': ['error', 4]
    }
};
```

</details>

**What did you do? Please include the actual source code causing the issue.**
I found `eslint-friendly-formatter` work weirdly, and I found it is a problem from `eslint`, and I've created a _Minimum Viable Project_: http://plnkr.co/edit/SKvcqNj8S3KiG4x7GRsL

Briefly, we specify a formatter for `eslint`. When a rule is violated by any code, the formatter will receive a `results` object containing error `messages`. Fields `column` and `source` in `results[i].messages[j]` are inconsistent, that is, the starting indent is ignored and dropped in `source` field, when `column` field is still original code position.

**What actually happened?**
Like this:

```html
<script>
    function a(){
        function b(){
        }
    }
</script>
```

This code violates `space-before-blocks` rule, then we'll get `source` field displaying as
```js
function a(){
```
rather than
```js
    function a(){
```
, while `column` field is 17, not 13.

The output from my _Minimun Viable Project_ can perfectly show this:

```plain
$ npm run lint

> eslint --format ./formatter.js index.html

{"column":17,"sourceLength":13,"source":"function a(){"}
{"column":21,"sourceLength":17,"source":"    function b(){"}
```

Here `sourceLength` is calculated as `source` field's length.
It will make some formatters work incorrectly, e.g. `eslint-friendly-formatter`:

```plain
  ✘  http://eslint.org/docs/rules/space-before-blocks  Missing space before opening brace  
  index.html:3:21
      function b(){
                      ^
```

The correct case should be:

```plain
  ✘  http://eslint.org/docs/rules/space-before-blocks  Missing space before opening brace  
  index.html:3:21
      function b(){
                  ^
```

**What did you expect to happen?**
plz make `column` and `source` fields to be consistent.

**Comment by mysticatea** [comment]
I guess this is a problem of eslint-plugin-html.
As far as I know, the plugin modifies the location of reports in hacked way.

Could you confirm that this issue is reproduced even if you remove the plugin?

**Comment by hsxfjames** [comment]
Hi @mysticatea
You are right. I was confused previously by how plugins work, but finally I find out that is a problem of eslint-plugin-html. It has a dedent procedure but calculates original location. I'll report this problem to eslint-plugin-html and this issue can be closed.
Sorry for troubling.

**Comment by hsxfjames** [comment]
But I still wonder if it is necessary for eslint-plugin-html to calculate original location or if it is needed by ESlint? If so, is it a problem of formatter not plugin? thx.

**Comment by not-an-aardvark** [comment]
I don't understand the question.

**Comment by hsxfjames** [comment]
Hi, @not-an-aardvark 
It seems that I asked a stupid question. I just don't know where should I report this problem, the formatter or the plugin, since I've confirmed that it's not a problem of ESLint itself. I'll continue to dive into it, thank you very much.
